code/exercises/camera_calibration/calibrated_lane_detector.py

```diff
@@ -0,0 +1,98 @@
+import numpy as np
+from ..lane_detection.lane_detector import LaneDetector
+from ..lane_detection.camera_geometry import CameraGeometry
+
+
+def get_intersection(line1, line2):
+    m1, c1 = line1
+    m2, c2 = line2
+    #TODO: find intersection of the line.
+    raise NotImplementedError
+
+def get_py_from_vp(u_i, v_i, K):
+    #TODO compute pitch and yaw given the camera intrinsic matrix and vanishing point.
+    raise NotImplementedError
+    return pitch, yaw
+
+class CalibratedLaneDetector(LaneDetector):
+    def __init__(self, calib_cut_v = 200, cam_geom=CameraGeometry(), model_path='./fastai_model.pth'):
+        # call parent class constructor
+        super().__init__(cam_geom, model_path)
+
+        self.calib_cut_v = calib_cut_v
+
+        # build u,v grid
+        uv = []
+        for v in range(calib_cut_v, self.cg.image_height):
+            for u in range(self.cg.image_width):
+                uv.append(np.array([u,v]))
+        self.uv_grid = np.array(uv)
+
+        self.estimated_pitch_deg = 0
+        self.estimated_yaw_deg = 0
+        self.update_cam_geometry()
+        self.mean_residuals_thresh = 2.5 #TODO: adjust this thresh hold to avoid calibration process at curves.

```

Comment:
Why 2.5? Should we maybe give something really large in the beginning, like 1e6 so that every image is being used? Then students can lower that number. If you think starting with a small number makes more sense, we can also leave it

Reply:
I didn't really give too much thought to this. But setting a high number looks better